torch/package/package_exporter.py

```diff
@@ -468,8 +492,23 @@ def add_dependency(self, module_name: str, dependencies=True):
 
                 # If we are interning this module, we need to retrieve its
                 # dependencies and package those as well.
-                if pattern_info.action == _ModuleProviderAction.INTERN:
+                if action == _ModuleProviderAction.INTERN:
                     self._intern_module(module_name, dependencies)
+                if action == _ModuleProviderAction.SELECTIVE_EXTERN:
+                    module_obj = self._import_module(module_name)
+                    is_package = hasattr(module_obj, "__path__")
+                    source = self._get_source_of_module(module_obj)
+
+                    assert is_package
```

Comment:
It's worth checking that the module is a package, but it should not be an assertion. AssertionError implies a bug or logical error (e.g. an invariant is violated in the code). In this case, a user could provide us with bad input to cause this to happen, so it should be a RuntimeError.

This has a practical effect too btw, if `__debug__` is not set, an AssertionError will not be raised.